testsuite/MDAnalysisTests/topology/test_itp.py

```diff
@@ -490,3 +490,20 @@ def test_missing_elements_no_attribute():
         u = mda.Universe(ITP_atomtypes)
     with pytest.raises(AttributeError):
         _ = u.atoms.elements
+
+
+def test_deprecation_warning():
+    "Test deprecation warning is present"
+    with pytest.warns(DeprecationWarning, match="removed in release 3.0"):
+        mda.Universe(ITP_nomass)
+
+
+def test_nodeprecation_warning():
+    "Test deprecation warning is not present if elements isn't guessed"
+    with pytest.warns(UserWarning) as record:
+        mda.Universe(ITP_atomtypes)
+    assert len(record) == 2
```

Comment:
It would be best to not do this - i.e. we know this is prone to failures as soon as someone adds another warning somewhere else. However it's best to just fix it later.

Reply:
You could also just use `no_deprecated_call` context manager I wrote for MDA 7 years ago in gh-1522.

Reply:
Thanks Tyler -- trying the below code raised an error for me that I won't have time to look at for a few hours. Any quick suggestions? Otherwise since @IAlibay has called the feature freeze for his Sunday, I can raise an issue and fix this afterwards as he suggested.

```
def test_elements_nodeprecation_warning():
    "Test deprecation warning is not present if elements isn't guessed"
    with no_deprecated_call():
        mda.Universe(ITP_atomtypes)
```

raises

```
../util.py:244: in __exit__
    if any(issubclass(c, deprecation_categories) for c in self._captured_categories):
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

.0 = <list_iterator object at 0x12cc235b0>

>   if any(issubclass(c, deprecation_categories) for c in self._captured_categories):
E   TypeError: issubclass() arg 1 must be a class

../util.py:244: TypeError
```

Reply:
Fair enough, analysis below, maybe I'll open a ticket about this since I think I agree with you that the explicit requirement to add `UserWarning` shouldn't be needed (though it does fix the problem). Perhaps this is some subtlety about Python 2 at the time vs. 3 now:

this works just fine:

```python
def test_elements_nodeprecation_warning():
    "Test deprecation warning is not present if elements isn't guessed"
    def foo():
        warnings.warn("boo", UserWarning)
    with no_deprecated_call():
        foo()

```

this fails like your report:

```python
def test_elements_nodeprecation_warning():
    "Test deprecation warning is not present if elements isn't guessed"
    def foo():
        warnings.warn("boo")
    with no_deprecated_call():
        foo()
```

this patch gets the new test form working as you might expect from above analysis:

<details>

```diff
--- a/package/MDAnalysis/core/universe.py
+++ b/package/MDAnalysis/core/universe.py
@@ -145,7 +145,7 @@ def _resolve_coordinates(filename, *coordinates, format=None,
             get_reader_for(filename, format=format)
         except (ValueError, TypeError):
             warnings.warn('No coordinate reader found for {}. Skipping '
-                            'this file.'.format(filename))
+                            'this file.'.format(filename), UserWarning)
         else:
             coordinates = (filename,) + coordinates
     return coordinates
diff --git a/package/MDAnalysis/topology/ITPParser.py b/package/MDAnalysis/topology/ITPParser.py
index d2f7b6da8..ff47f0728 100644
--- a/package/MDAnalysis/topology/ITPParser.py
+++ b/package/MDAnalysis/topology/ITPParser.py
@@ -607,7 +607,7 @@ class ITPParser(TopologyReaderBase):
             warnings.warn("Element information is missing, elements attribute "
                           "will not be populated. If needed these can be "
                           "guessed using universe.guess_TopologyAttrs("
-                          "to_guess=['elements']).")
+                          "to_guess=['elements']).", UserWarning)
```

</details>